### DISCUSSION TITLE
Rename filename in UI

### DISCUSSION THREAD

**Discussion by matzeeable** [WeblateOrg/weblate#6537] [Ideas] [answered]
Hey there!

### Context

- Currently using the [Local files](https://docs.weblate.org/en/latest/vcs.html#local-files) mode for all our projects and components
- All our projects and components are configured with the `gettext (PO)` file format
- File mask: `real-category-library-*.po`
- Style of language codes: Depending on file format -> changing to something like `POSIX with underscore` does not change the below behavior

### Current solution

At the moment, if we create a new language within our component, the filename is automatically `real-category-library-es.po`:

![image](https://user-images.githubusercontent.com/1008534/132701633-d8a3f5df-55ea-4157-ad02-82af918eed82.png)

But our application needs to download the file as `real-category-library-es_ES.po` (WordPress plugin). So, we are doing the following workaround:

```bash
ssh max@weblate.example.com
sudo docker exec -it weblate /bin/bash # This is our docker container
cd /app/data/vcs/{project}/{component}
mv real-category-library-es.po real-category-library-es_ES.po
git add -A
git commit -m "Renamed translation file using SSH (Spanish)"
```

Afterwards, we need to let Weblate know about this change and navigate to `https://weblate.example.com/projects/wordpress-real-category-management-backend-php/ci-mojito/es/#repository` and click on `Rebase`.

### Expected solution

A proposal would be to change the filename within the "Information" tab besides the "Download" button or define the filename when starting a new translation.

**EDIT:** Or is there another feature we need to use to define the filename? I just saw that we are using the `weblate.consistency.languages` addon, and we need to use the modified filenames for the other components, too.

What do you think?

Regards,
Matthew 😊

**Comment by matzeeable** [comment, marked as answer]
I just found the solution. Instead of using the simple `POSIX with underscore` language code style, I need to use the `POSIX with underscore, including country code`. (`language_code_style=posix_long`)

**Comment by nijel** [comment]
You can also define custom mapping for the language codes using https://docs.weblate.org/en/latest/admin/projects.html#language-aliases

**Reply by matzeeable** [reply]
Hey @nijel ! Where do I find this option in the UI?

**Reply by matzeeable** [reply]
Ok, I found the input in the UI in `https://weblate.example.com/settings/{project}/#workflow`, but what do I need to do to change the `real-category-library-fr.po` filename to `real-category-library-fr_FR.po`?

**Reply by nijel** [reply]
fr:fr_FR, but using the language code choices as you've posted in previous post is better solution to this particular problem.
